From f442357fadb535f4f343dc0593adb945fa03fc0c Mon Sep 17 00:00:00 2001
From: Josh Wu <josh.wu@atmel.com>
Date: Wed, 3 Sep 2014 15:16:14 +0800
Subject: [PATCH] ARM: at91sam9rlek: update the nand flash mapping and boot
 command & args

Align the nand flash map to linux4sam.org

Signed-off-by: Josh Wu <josh.wu@atmel.com>
---
 include/configs/at91sam9rlek.h | 16 +++++++++-------
 1 file changed, 9 insertions(+), 7 deletions(-)

diff --git a/include/configs/at91sam9rlek.h b/include/configs/at91sam9rlek.h
index d221ea1..be4d814 100644
--- a/include/configs/at91sam9rlek.h
+++ b/include/configs/at91sam9rlek.h
@@ -163,14 +163,16 @@
 
 /* bootstrap + u-boot + env + linux in nandflash */
 #define CONFIG_ENV_IS_IN_NAND		1
-#define CONFIG_ENV_OFFSET		0x60000
-#define CONFIG_ENV_OFFSET_REDUND	0x80000
+#define CONFIG_ENV_OFFSET		0xc0000
+#define CONFIG_ENV_OFFSET_REDUND	0x100000
 #define CONFIG_ENV_SIZE		0x20000		/* 1 sector = 128 kB */
-#define CONFIG_BOOTCOMMAND	"nand read 0x22000000 0xA0000 0x200000; bootm"
-#define CONFIG_BOOTARGS		"console=ttyS0,115200 " \
-				"root=/dev/mtdblock5 " \
-				"mtdparts=atmel_nand:128k(bootstrap)ro,256k(uboot)ro,128k(env1)ro,128k(env2)ro,2M(linux),-(root) " \
-				"rw rootfstype=jffs2"
+#define CONFIG_BOOTCOMMAND	"nand read 0x22000000 0x200000 0x300000; bootz 0x22000000"
+#define CONFIG_BOOTARGS		\
+				"console=ttyS0,115200 earlyprintk "				\
+				"mtdparts=atmel_nand:256k(bootstrap)ro,512k(uboot)ro,"		\
+				"256K(env),256k(evn_redundent),256k(spare),"			\
+				"512k(dtb),6M(kernel)ro,-(rootfs) "				\
+				"rootfstype=ubifs ubi.mtd=7 root=ubi0:rootfs"
 
 #else /* CONFIG_SYS_USE_MMC */
 
-- 
1.9.1

